hasattr(faker, 'bank') returns True for en_US locale

Problem occurs from version 37.12.0

```python
def test_faker():
    locale = 'en_US'
    faker = Faker(locale=locale)
    faker.seed_instance(123)

    if hasattr(faker, 'company'): # True
        value = faker.company() # Ok
        print(value) # 'Adams Group'

    if hasattr(faker, 'bank'): # True
        value = faker.bank() # AttributeError: The Provider provider does not have a 'banks' attribute. Consider contributing to the project and  adding a 'banks' tuple to enable bank name generation.
        print(value)
```
full error message
```
self = <faker.providers.bank.en_GB.Provider object at 0x7f7955a88190>

    def bank(self) -> str:
        """Generate a bank name."""
        if not hasattr(self, "banks"):
>           raise AttributeError(
                f"The {self.__class__.__name__} provider does not have a 'banks' "
                "attribute. Consider contributing to the project and "
                " adding a 'banks' tuple to enable bank name generation."
            )
E           AttributeError: The Provider provider does not have a 'banks' attribute. Consider contributing to the project and  adding a 'banks' tuple to enable bank name generation.

.venv/lib/python3.14/site-packages/faker/providers/bank/__init__.py:54: AttributeError
```

--- a/faker/providers/bank/__init__.py
+++ b/faker/providers/bank/__init__.py
@@ -51,7 +51,7 @@ class Provider(BaseProvider):
     def bank(self) -> str:
         """Generate a bank name."""
         if not hasattr(self, "banks"):
-            raise AttributeError(
+            raise NotImplementedError(
                 f"The {self.__class__.__name__} provider does not have a 'banks' "
                 "attribute. Consider contributing to the project and "
                 " adding a 'banks' tuple to enable bank name generation."

--- a/tests/providers/test_bank.py
+++ b/tests/providers/test_bank.py
@@ -64,13 +64,21 @@ class TestBaseBankProvider:
     """Test base bank provider"""
 
     def test_bank_not_implemented_error(self, faker):
-        """Test that bank() raises AttributeError when no banks attribute exists"""
+        """Test that bank() raises NotImplementedError when no banks attribute exists.
+
+        Raising NotImplementedError (instead of AttributeError) ensures that
+        hasattr(faker, 'bank') correctly returns True (the method exists) while
+        still clearly communicating that this locale has not implemented bank
+        name generation. Previously AttributeError was raised, which caused
+        hasattr() to return False-positive True yet crash on the call.
+        See: https://github.com/joke2k/faker/issues/2377
+        """
 
         provider = BankProvider(faker)
 
         assert not hasattr(provider, "banks")
 
-        with pytest.raises(AttributeError):
+        with pytest.raises(NotImplementedError):
             provider.bank()
 
 

